pkg/cmd/cli/cmd/set/env.go

```diff
@@ -115,6 +119,133 @@ func keyToEnvName(key string) string {
 	return strings.ToUpper(validEnvNameRegexp.ReplaceAllString(key, "_"))
 }
 
+func getRefObject(f *clientcmd.Factory, reference string) (runtime.Object, error) {
```

Comment:
this is overkill... just use the kclient to look up the configmap or secret object directly in getSecretRefValue or getConfigMapRefValue, no need to go through resource builder